engines/lilliput/lilliput.cpp

```diff
@@ -0,0 +1,2897 @@
+	for (int i = 0; i < _interfaceHotspotNumb; i++) {
+		if (_scriptHandler->_interfaceButtonActivationDelay[i] != 0) {
+			--_scriptHandler->_interfaceButtonActivationDelay[i];
+			if (_scriptHandler->_interfaceButtonActivationDelay[i] == 0) {
+				_scriptHandler->_interfaceHotspotStatus[i] = kHotspotEnabled;
+				needRedraw = true;
+			}
+		}
+	}
+
+	if (needRedraw)
+		displayInterfaceHotspots();
+}
+
+void LilliputEngine::displayHeroismIndicator() {
+	debugC(2, kDebugEngine, "displayHeroismIndicator()");
+
+	if (_scriptHandler->_savedBuffer215Ptr == NULL)
+		return;
+
+	int var1 = (_scriptHandler->_savedBuffer215Ptr[0] * 25) >> 8;
+
+	if (var1 == _scriptHandler->_heroismLevel)
+		return;
+
+	int var2 = 1;
+	if (var1 > _scriptHandler->_heroismLevel)
+		var1 = 150;
+	else {
+		var2 = -1;
+		var1 = 40;
+	}
+
+	_scriptHandler->_heroismLevel += var2;
+
+	int index = _scriptHandler->_heroismBarX + (_scriptHandler->_heroismBarBottomY * 320);
+
+	var2 = _scriptHandler->_heroismLevel & 0xFF;
+	if (var2 != 0) {
+//		sub16064(var1, _scriptHandler->_byte15FFA);
+		for (int i = 0; i < (var2 << 2); i++) {
+			((byte *)_mainSurface->getPixels())[index] = var1;
+			((byte *)_mainSurface->getPixels())[index + 1] = var1;
+			((byte *)_mainSurface->getPixels())[index + 2] = var1;
+			index -= 320;
+		}
+	}
+
+	if (25 - _scriptHandler->_heroismLevel != 0) {
+//		sub16064(23, 25 - _scriptHandler->_byte15FFA);
+		var2 = (25 - _scriptHandler->_heroismLevel) << 2;
+		for (int i = 0; i < var2; i++) {
+			((byte *)_mainSurface->getPixels())[index] = 23;
+			((byte *)_mainSurface->getPixels())[index + 1] = 23;
+			((byte *)_mainSurface->getPixels())[index + 2] = 23;
+			index -= 320;
+		}
+	}
+}
+
+void LilliputEngine::pollEvent() {
+	debugC(2, kDebugEngine, "pollEvent()");
+
+	Common::Event event;
+	while (_system->getEventManager()->pollEvent(event)) {
+		switch (event.type) {
+		case Common::EVENT_MOUSEMOVE:
+		case Common::EVENT_LBUTTONDOWN:
+		case Common::EVENT_LBUTTONUP: {
+			Common::Point newMousePos = Common::Point(CLIP<int>(event.mouse.x, 0, 304), CLIP<int>(event.mouse.y, 0, 184));
+
+			if (_mousePreviousEventType != event.type) {
+				_mousePreviousEventType = event.type;
+				if (_mouseButton != 1) {
+					_mouseButton = 2;
+					if (event.type != Common::EVENT_MOUSEMOVE) {
+						_mouseButton = 1;
+						_mousePos = Common::Point(newMousePos.x + 5, newMousePos.y + 1);
+					}
+				} else {
+					_mouseClicked = true;
+				}
+			}
+
+			if (newMousePos != _oldMousePos) {
+				_oldMousePos = newMousePos;
+				_mouseDisplayPos = newMousePos;
+			}
+			_lastEventType = event.type;
+			}
+			break;
+		case Common::EVENT_QUIT:
+			_shouldQuit = true;
+			break;
+		case Common::EVENT_KEYUP:
+		case Common::EVENT_KEYDOWN: {
+			if ((event.type == _lastKeyPressed.type) && (event.kbd == _lastKeyPressed.kbd))
+				break;
+
+			_lastKeyPressed = event;
+			int nextIndex = (_keyboard_nextIndex + 1) % 8;
+			if (_keyboard_oldIndex != nextIndex) {
+				_
```

Comment:
Setting the seed to a fixed value will make the random source always return the same sequence.

Reply:
@bgK I believe you meant to comment on another line? :)

Reply:
EDIT: If someone else is wondering, this regards line 2847. For some reason the preview points to the wrong piece of code.